client.go

```diff
@@ -1354,6 +1354,12 @@ var clientConnPool sync.Pool
 
 func (c *HostClient) releaseConn(cc *clientConn) {
 	cc.lastUseTime = time.Now()
+	//reset read and write deadline for reuse
+	if cc.c != nil {
+		cc.c.SetReadDeadline(time.Time{})
+		cc.c.SetWriteDeadline(time.Time{})
+	}
```

Comment:
Thanks for your response. 
Do you mean this?
<img width="342" alt="screen shot 2018-11-19 at 6 46 09 pm" src="https://user-images.githubusercontent.com/1196476/48702266-737b2800-ec2b-11e8-83ed-ed19d2f702da.png">

I think this will make new request to set the new deadline, which will cause the performance issue again. 
I tried this and `SetReadDeadline` was called every time.

Reply:
Yes like that. I think it's impossible to avoid having each new request reset the deadline. The only way to avoid having to call SetDeadline again is when the idle timeout is less than the read timeout and the connection wasn't idle at all. As soon as the connection goes idle the current deadline is wrong and you get the problem this pull request is about. If the deadline is lower than the idle timeout the connection will timeout while being idle which also causes problems when we try to use it again.

Reply:
I'm looking at the issue linked in the comment and it actually seems like it has been fixed newer versions of Go: https://github.com/golang/go/issues/15133#issuecomment-271571395

I think it's best if we completely remove this optimization and just always call SetDeadline when needed.

Reply:
Great! I've done it. :)